src/smvlang/expr2smv.cpp

```diff
@@ -21,33 +21,75 @@ class expr2smvt
   {
   }
 
-  bool convert_nondet_choice(const exprt &src, std::string &dest, unsigned precedence);
+protected:
+  // In NuSMV 2.6., ! (not)  has a high precedence (above ::), whereas
+  // in the CMU SMV implementation it has the same as other boolean operators.
+  // We use the CMU SMV precedence.
+  enum class precedencet
+  {
+    MAX = 17,
+    INDEX = 16,  // [ ] , [ : ]
+    CONCAT = 15, // ::
+    UMINUS = 14, // - (unary minus)
+    MULT = 13,   // * / mod
+    PLUS = 12,   // + -
+    SHIFT = 11,  // << >>
+    UNION = 10,  // union
+    IN = 9,      // in
+    REL = 8,     // = != < > <= >=
+    TEMP = 7,    // AX, AF, etc.
```

Comment:
Perhaps spell out "TEMPORAL" on this occasion? When reading the code I found myself thinking "wait, is there some sort of _temporary_ precedence?"